modules/contact/src/constraints/ComputeDynamicFrictionalForceLMMechanicalContact.C

```diff
@@ -59,6 +66,16 @@ ComputeDynamicFrictionalForceLMMechanicalContact::ComputeDynamicFrictionalForceL
       "in order to make its implementation feasible");
 #endif
 
+  if (!_has_friction_function && !isParamValid("mu"))
+    paramError(
+        "mu",
+        "A coefficient of friction needs to be provided as a constant value of via a function.");
```

Comment:
My understanding of `paramError` is that it will give the line number corresponding to the problematic parameter. However, if the parameter doesn't actually exist, I don't know what it does. I don't know if it will error itself, emit unknown and undefined dragons, or work very peacefully.

Reply:
It seems to behave like a regular error in that case. Switching to `mooseError`.